### PR TITLE
feat(organizations): add settings page for editing and deleting orgs

### LINKED ISSUE #7611: Feature: Add organization settings (edit) page
### Is your feature request related to a problem? Please describe

Organizations currently have no way to edit their details after creation. There's no UI to update the name, description, location, visibility, logo, or external links, and no way to delete an organization from its own page.

### Describe the solution you'd like

Add a settings/edit page for organizations where an admin can:
- update the name, description, location, and visibility
- upload/replace the logo
- add or remove external links (up to 5)
- see a live slug preview + availability check while editing the name

Also add a "Danger Zone" on this page to delete the organization, with a type-to-confirm modal so it can't be triggered accidentally.

This is part of the Organizations feature and depends on the controller PR (#7457) for the edit/update/destroy actions and #7568.

### Describe alternatives you've considered

No alternative 

### Additional context

Part of project-6 Enterprise & Institutional Organization Features.

### Are you working on this?

Yes

### PR DESCRIPTION
**Pull request by naman79820**
Fixes #7611

#### Describe the changes you have made in this PR -
This PR adds the organization settings page, allowing organization admins to edit their organization's details and delete the organization.

Admins can update the name, description, location, visibility, external links (up to 5, with automatic provider-logo detection), and logo. The page also includes a "Danger Zone" for deleting the organization, protected by a type-the-name-to-confirm step that is enforced on the server.

The form is built as a reusable OrganizationFormComponent (ViewComponent) so it can be reused for organization creation in a follow-up PR. A Lookbook preview is included for the component.
### Screenshots of the UI changes (If any) -

https://github.com/user-attachments/assets/220e523c-5b5f-4c07-afd2-9bcb3961c495

---

## Code Understanding and AI Usage

**Did you use AI generated code (ChatGPT, Claude, Copilot, etc.) in any part of this PR? **
- [x] No, I wrote all the code myself
- [ ] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [ ] I have reviewed every single line of the AI-generated code
- [ ] I can explain the purpose and logic of each function/component I added
- [ ] I have tested edge cases and understand how the code handles them
- [ ] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**
Organizations could be created but not edited or deleted. This adds a settings page where admins can update the organization's details (name, description, location, visibility, external links, logo) and delete it.

For the delete confirmation, I considered a client-side "type the name to enable the button" check, but made it server-side instead  a disabled button can be bypassed by disabling JS or sending a direct request, so destroy verifies the typed name matches on the server before deleting.

For authorization, instead of the shell component calling Pundit directly, the policy result is computed in the view and passed in as an input, so the component just renders what it's told (and can be previewed in Lookbook without a Pundit context).

I built the form as a reusable ViewComponent since the same form will be reused for organization creation in a follow-up PR.

Organizations are moving to UUID-based identifiers in a follow-up PR. As a step toward that, this PR removes the slug preview UI from the form and keeps the slug stable when an organization is renamed (renaming updates the name but not the URL). The check_slug endpoint and related JS/locale keys are retained but unused for now, and will be removed together with the rest of the slug logic in the UUID PR.

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

## Summary by CodeRabbit

* **New Features**
  * Added organization dashboard navigation with Overview, Members (placeholder), and Settings tabs.
  * Added organization overview pages with group cards, member counts, pagination, and social links.
  * Added organization settings UI with improved external link accessibility, logo upload UX, and guarded delete confirmation.

* **Bug Fixes**
  * Improved organization link security/validation (HTTP/HTTPS only, automatic scheme handling, max links).
  * Updated redirects to always land on the Overview tab after create/update.
  * Prevented organization deletion unless the confirmation matches and permissions allow it.

### CODE REVIEW

File: app/component/organization_form_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    <label class="form-label fw-semibold" for="organization_name">
+      <%= t("organizations.settings.name_label") %> <span class="text-danger">*</span>
+    </label>
+    <div class="form-text mt-0 mb-2"><%= t("organizations.settings.name_hint") %></div>
+    <div class="input-group">
+      <span class="input-group-text"><i class="fa fa-building org-form-icon" aria-hidden="true"></i></span>
+      <%= form.text_field :name, id: "organization_name", class: "form-control",
+            placeholder: t("organizations.settings.name_placeholder"), required: true, maxlength: 50 %>
+    </div>
+    <div class="text-end"><small id="org-name-counter" class="form-text"></small></div>
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/component/organization_form_component.html.erb:32]: I removed it because we're going to use a UUID instead of a humanized slug. I'll implement that in an upcoming PR, which is why I removed the preview. As for all the slug logic, I'll remove it in that same PR :)) @coderabbitai

File: app/controllers/organizations_controller.rb [hunk 2]
```diff
@@ -3,38 +3,73 @@
...
   end
 
-  # GET /organizations/1/edit
-  def edit; end
+  # GET /organizations/check_slug
+  def check_slug
+    base_slug = (params[:slug].presence || params[:name]).to_s.strip.parameterize
+    is_taken = base_slug.present? && Organization.exists?(slug: base_slug)
+
+    render json: { slug: base_slug, available: base_slug.present? && !is_taken }
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/controllers/organizations_controller.rb:63]: The slug preview UI has been removed from the form (organizations are moving to UUID identifiers in a follow-up PR), so check_slug is no longer called. The endpoint and this logic will be removed entirely in that PR, so I've left it as-is rather than fixing a path that's being deleted.

File: app/javascript/controllers/organization_form_controller.js [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+        const counter = document.getElementById(counterId);
+        if (!el || !counter) return;
+        const update = () => {
+            const remaining = maxLen - el.value.length;
+            counter.textContent = `${remaining} characters remaining`;
+            counter.classList.toggle('text-danger', remaining <= Math.floor(maxLen * 0.15));
+        };
+        update();
+        el.addEventListener('input', update);
+    }
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/javascript/controllers/organization_form_controller.js:30]: Added /* eslint-disable class-methods-use-this */ at the top of the controller, matching the existing pattern in notifications_controller.js, projects_controller.js, and assignment_controller.js

File: app/javascript/controllers/organization_form_controller.js [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+                    .then((r) => r.json())
+                    .then((data) => {
+                        if (requestId !== slugRequestId) return;
+                        slugPreview.textContent = data.slug || currentSlug;
+                        setSlugStatus((data.available || data.slug === currentSlug) ? 'available' : 'taken', data.slug);
+                    })
+                    .catch(() => {
+                        if (requestId === slugRequestId) setSlugStatus('idle', '');
+                    });
+            }, 350);
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/javascript/controllers/organization_form_controller.js]: i was thinking to remove whole logic in different pr but removing setupslug in this pr itself

File: app/views/organizations/members.html.erb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<% content_for :title, t("organizations.members.page_title", organization: @organization.name) %>
+
+<%= render(OrganizationDashboardShellComponent.new(organization: @organization, active_tab: "members", show_settings_tab: policy(@organization).admin_access?)) do |shell| %>
+  <% shell.with_tab_content do %>
+    <div class="org-dash-empty">
+      <i class="fa fa-users" aria-hidden="true"></i>
+      <p>Members management is coming soon.</p>
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/views/organizations/members.html.erb:7]: lets keep it like this as we will change it later on anyway

File: app/assets/stylesheets/organizations.scss [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+  text-align: center;
+  cursor: pointer;
+  transition: border-color 0.15s, background-color 0.15s;
+}
+
+.org-upload-zone:hover,
+.org-upload-zone--active {
+  border-color: var(--cv-primary-green);
+  background-color: var(--cv-light-green-bg);
+}
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/assets/stylesheets/organizations.scss:158]: recheck it is resolved @coderabbitai

File: app/controllers/organizations_controller.rb [hunk 7]
```diff
@@ -45,22 +80,25 @@ def create
...
     end
   end
 
   # DELETE /organizations/1
-  # DELETE /organizations/1.json
   def destroy
+    if params[:confirmation] != @organization.name
+      redirect_to settings_organization_path(@organization), alert: t(".confirmation_mismatch")
+      return
+    end
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/controllers/organizations_controller.rb:103]: recheck @coderabbitai

File: app/views/organizations/settings.html.erb [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+              <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="<%= t('close') %>"></button>
+            </div>
+            <div class="modal-body p-4">
+              <div class="alert alert-danger py-2 small">
+                <%= t("organizations.settings.modal_warning", organization: @organization.name) %>
+              </div>
+              <p class="small mb-2"><%= t("organizations.settings.modal_confirm_prompt_html", organization: @organization.name) %></p>
+              <%= form_with url: organization_path(@organization), method: :delete, local: true do %>
+                <input type="text" id="deleteOrgConfirmInput" name="confirmation" class="form-control mb-3" autocomplete="off"
+                      data-org-name="<%= @organization.name %>">
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/views/organizations/settings.html.erb:49]: recheck @coderabbitai
